tests/netflix/test_cassandra_uncertain.py

```diff
@@ -150,72 +154,61 @@ def test_worn_dataset():
+            ),
+            actual_attached_disk=lr_cluster.attached_drives[0]
+            if lr_cluster.attached_drives
+            else None,
+        )
 
 
-def test_worn_dataset_ebs():
+def test_worn_dataset_force_ebs():
     """Assert that a write once read never (aka tracing) dataset uses
-    CPU and GP2 cloud drives to max ability. Paying for fast ephmeral storage
-    is silly when we're never reading from it.
+    has reasonable outputs for ebs results
     """
-    worn_desire = CapacityDesires(
-        service_tier=1,
-        query_pattern=QueryPattern(
-            # Very Very few reads.
-            estimated_read_per_second=Interval(
-                low=1, mid=10, high=100, confidence=0.98
-            ),
-            # We think we're going to have around 1 million writes per second
-            estimated_write_per_second=Interval(
-                low=100_000, mid=1_000_000, high=2_000_000, confidence=0.98
-            ),
-        ),
-        # We think we're going to have around 200 TiB of data
-        data_shape=DataShape(
-            estimated_state_size_gib=Interval(
-                low=104800, mid=204800, high=404800, confidence=0.98
-            ),
-        ),
-    )
     cap_plan = planner.plan(
         model_name="org.netflix.cassandra",
         region="us-east-1",
         desires=worn_desire,
         extra_model_arguments={
             "max_regional_size": 200,
             "copies_per_region": 2,
-            "require_local_disks": False,
             "require_attached_disks": True,
         },
     )
 
-    lr = cap_plan.least_regret[0]
-    lr_cluster = lr.candidate_clusters.zonal[0]
-    assert 128 <= lr_cluster.count * lr_cluster.instance.cpu <= 512
-    assert (
-        250_000
-        <= lr.candidate_clusters.annual_costs["cassandra.zonal-clusters"]
-        < 1_000_000
-    )
-    assert lr_cluster.instance.family in ("m5", "r5", "r6a")
-    assert lr_cluster.attached_drives[0].name == "gp3"
-    # gp2 should not provision massive drives, prefer to upcolor
-    assert lr_cluster.attached_drives[0].size_gib < 9000
-    assert lr_cluster.attached_drives[0].size_gib * lr_cluster.count * 3 > 204800
-    # We should have S3 backup cost
-    assert lr.candidate_clusters.services[0].annual_cost > 5_000
+    lr_clusters = [lr.candidate_clusters.zonal[0] for lr in cap_plan.least_regret]
+    for lr_cluster in lr_clusters:
+        assert has_attached_storage(lr_cluster)
+        assert lr_cluster.attached_drives[0].name == "gp3"
+        assert_minimum_storage_gib(lr_cluster, 102_400)
+        # gp3 should not provision massive drives, prefer to upcolor
+        assert get_drive_size_gib(lr_cluster) <= 8 * 1024
+
+        # (matthewho) Why is this r6a.2xlarge??? The above test has the
+        # same desires and is m7a.xlarge. The only meaningful difference is
+        # the required_attached_disks. Also, the resolved disk size is larger
+        assert_similar_compute(
```

Comment:
This output is strange. The desires are the same as the previou stest but somehow the output is r6a.2xlarge (as opposed to r7a.xlarge).

It's hard to walk through the code in the uncertain code to understand why the outputs are different at all. I timeboxed this change short-term but would love some extra eyes to understand this behvaior better.